xarray/core/dataarray.py

```diff
@@ -3866,7 +3870,10 @@ def to_pandas(self) -> Self | pd.Series | pd.DataFrame:
                 "pandas objects. Requires 2 or fewer dimensions."
             ) from err
         indexes = [self.get_index(dim) for dim in self.dims]
-        return constructor(self.values, *indexes)  # type: ignore[operator]
+        pandas_object = constructor(self.values, *indexes)
```

Comment:
let's handle the extension array special-case here using `isinstance(self._variable._data, ExtensionArray)`

Reply:
Great!